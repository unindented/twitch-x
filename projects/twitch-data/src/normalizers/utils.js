export function valueOrNull (value) {
  return value || null
}
