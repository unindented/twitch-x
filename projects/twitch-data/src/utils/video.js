import {stringify} from 'qs'
import {USHER_API_URL} from '../constants'

export function getPlaylistUrl ({stream, sig, token}) {
  const query = {
    sig,
    token,
    allow_audio_only: true,
    allow_source: true,
    player: 'twitchweb',
    type: 'any',
    p: Math.floor(Math.random() * 1e7)
  }
  return `${USHER_API_URL}/channel/hls/${stream}.m3u8?${stringify(query)}`
}

export function extractFeedsFromPlaylist (input) {
  const regexp = /#EXT-X-MEDIA:.*NAME="([^"]+)".*\n#EXT-X-STREAM-INF:.*\n(.*)/g
  const result = []

  let match = regexp.exec(input)
  while (match != null) {
    result.push({
      name: match[1],
      url: match[2]
    })
    match = regexp.exec(input)
  }

  return result
}
