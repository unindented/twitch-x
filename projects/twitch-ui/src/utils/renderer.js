import React from 'react'
import renderer from 'react-test-renderer'
import {ThemeProvider} from 'styled-components'
import themes from '../themes'

export function renderWithTheme (node, theme = 'default') {
  return renderer.create(
    <ThemeProvider theme={themes[theme]}>
      {node}
    </ThemeProvider>
  )
}
