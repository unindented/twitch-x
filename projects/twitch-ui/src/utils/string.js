export function supplant (str, obj) {
  return str.replace(/\{\s*([^{}]+?)\s*\}/g, (str, key) => obj[key])
}
