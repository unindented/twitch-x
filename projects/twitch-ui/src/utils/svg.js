export function rectangle ({width, height, fill}) {
  return `<svg viewBox="0 0 ${width} ${height}" xmlns="http://www.w3.org/2000/svg"><rect fill="${fill}" width="100%" height="100%" /></svg>`
}
