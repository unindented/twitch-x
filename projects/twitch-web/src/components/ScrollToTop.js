import {PureComponent} from 'react'
import PropTypes from 'prop-types'
import {withRouter} from 'react-router'

class ScrollToTop extends PureComponent {
  componentDidUpdate (prevProps) {
    if (this.props.location !== prevProps.location) {
      window.scrollTo(0, 0)
    }
  }

  render () {
    return this.props.children
  }
}

ScrollToTop.propTypes = {
  location: PropTypes.object.isRequired,
  children: PropTypes.node
}

export default withRouter(ScrollToTop)
