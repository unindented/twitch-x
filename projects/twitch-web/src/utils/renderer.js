import ShallowRenderer from 'react-test-renderer/shallow'

export function render (node, theme = 'default') {
  const renderer = new ShallowRenderer()
  renderer.render(node)
  return renderer.getRenderOutput()
}
