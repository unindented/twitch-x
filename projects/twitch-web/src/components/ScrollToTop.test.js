import React from 'react'
import {render} from '../utils/renderer'
import ScrollToTop from './ScrollToTop'

describe('ScrollToTop', () => {
  let tree

  beforeEach(() => {
    tree = render(
      <ScrollToTop>
        foo
      </ScrollToTop>
    )
  })

  it('renders', () => {
    expect(tree).toMatchSnapshot()
  })
})
