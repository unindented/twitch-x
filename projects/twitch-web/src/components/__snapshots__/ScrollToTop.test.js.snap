// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTop renders 1`] = `
<Route
  render={[Function]}
/>
`;
